src/backend/cpu/blas.cpp

```diff
@@ -286,29 +307,36 @@ Array<T> matmul(const Array<T> &lhs, const Array<T> &rhs, af_mat_prop optLhs,
             const MKL_INT ldb = rStrides[1];
             const MKL_INT ldc = oStrides[1];
 
-            gemm_batch_func<T>()(CblasColMajor, &lOpts, &rOpts, &M, &N, &K,
-                                 &alpha, lptrs.data(), &lda, rptrs.data(), &ldb,
-                                 &beta, optrs.data(), &ldc, 1, &batchSize);
+            gemm_batch_func<T>()(CblasColMajor, &lOpts, &rOpts,
+                                 &M, &N, &K,
+                                 alpha_batched.getScale(),
+                                 lptrs.data(), &lda, rptrs.data(), &ldb,
+                                 beta_batched.getScale(),
+                                 optrs.data(), &ldc, 1, &batchSize);
 #else
             for (int n = 0; n < batchSize; n++) {
                 if (rDims[bColDim] == 1) {
                     dim_t incr =
                         (optRhs == AF_MAT_NONE) ? rStrides[0] : rStrides[1];
-                    gemv_func<T>()(CblasColMajor, lOpts, lDims[0], lDims[1],
-                                   alpha, lptrs[n], lStrides[1], rptrs[n], incr,
-                                   beta, optrs[n], 1);
+                    gemv_func<T>()(CblasColMajor, lOpts,
+                                   lDims[0], lDims[1],
+                                   alpha_.getScale(),
+                                   lptrs[n], lStrides[1], rptrs[n], incr,
+                                   beta_.getScale(),
+                                   optrs[n], oStrides[0]);
                 } else {
-                    gemm_func<T>()(CblasColMajor, lOpts, rOpts, M, N, K, alpha,
+                    gemm_func<T>()(CblasColMajor, lOpts, rOpts,
+                                   M, N, K,
+                                   alpha_.getScale(),
```

Comment:
Shouldn't the `alpha_` and `beta_` change for each iteration ? or does the `gemm_func` correctly picks ith batch set's alpha and beta accordingly ?

Reply:
Umar and I originally planned to be like that, where alpha and beta could be different for each slice in batching. However, cuBLAS's batched gemm couldn't handle that case (see https://docs.nvidia.com/cuda/cublas/index.html#cublas-lt-t-gt-gemmbatched). Thus, to maintain behavior parity across all backends, we decided that we should just have all backends use a single alpha and beta for all batched slices.